src/arrays/arrays_with_methods/hard/q6/MainTest.java

```diff
@@ -0,0 +1,97 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    public static Stream<Integer> randomReplaceInput() {
+        return Stream.of(3, 4, 5, 6, 7, 40);
+    }
+
+    public static Stream<Arguments> oddIndexInput(){
+        return Stream.of(
+                Arguments.of(new int[]{1, 38, 42, 35, 5, 124, 4352, 54, 8, 43245, 79}, new int[]{38, 35, 124, 54, 43245}),
+                Arguments.of(new int[]{2, 43, 342, 5, 324, 543, 543, 6, 43, 634, 49, 236, 246, 2673653, 7653}, new int[]{43, 5, 543, 6, 634, 236, 2673653}),
+                Arguments.of(new int[]{10, 8, 6, 4, 2, 0}, new int[]{8, 4, 0})
+        );
+    }
+
+    @MethodSource("oddIndexInput")
+    @ParameterizedTest
+    public void randomReplaceMethodKeepsOddIndexNumber(int[] arr, int[] solutions) throws Throwable {
+        //this checks that the odd index numbers are not modified
+        Object[][] arguments = {
+                {arr, int[].class}
+        };
+        MethodTest m = new MethodTest(ShrinkItDown.class, "randomReplace", arguments);
+        if (m.getMethodReturnType() != int[].class) {
+            Assertions.fail("Your createRandomArray method does not return an array of integers.");
+        }
+        int[] output = (int[]) m.callMethod();
+        int[] oddNums = new int[output.length/2];
+        int count = 0;
+        for (int x = 0; x < output.length; x++) {
+            if (x % 2 == 1)
+                oddNums[count++] = output[x];
+        }
+        CustomAssertions._assertArrayEquals(solutions, oddNums, "Your program does not modify only the even indexed elements.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("randomReplaceInput")
+    public void correctRandomReplaceMethod(int size) throws Throwable {
+        //generates the input array
+        int[] arr = ArrayUtil.generateAscendingArray(1, size);
+        //an array of the arrays storing the generated numbers at each even index
+        int[][] storeRandoms = new int[(size + 1) / 2][1000];
+
+        Object[][] arguments = {
+                {arr, int[].class}
+        };
+        MethodTest m = new MethodTest(ShrinkItDown.class, "randomReplace", arguments);
+        if (m.getMethodReturnType() != int[].class) {
+            Assertions.fail("Your createRandomArray method does not return an array of integers.");
+        }
+
+        int repetitions = 0;
+
+        //runs the program 1000 times and stores all the even index numbers generated. Each are in their own array
+        //because each index has its own range the variables can fall in (I'm effectively using ascending array to control that)
+        for (int x = 0; x < 1000; x++) {
+            for (int y = 0; y < size; y++) {
+                arr[y] = y + 1;
+            }
+            int count = 0;
+            int[] output = (int[]) m.callMethod();
```

Comment:
If they don't return an array, will this not crash the program?

Reply:
nope, the above
if (m.getMethodReturnType() != int[].class) {
            Assertions.fail("Your createRandomArray method does not return an array of integers.");
        }
is to make sure that if they don't return an int[] that the test fails before it crashes. Just tested it by making the method return void and the tests just returned a lot of "Your createRandomArray method does not return an array of integers." with no crashes